Add bootstrap for SS locations

lower29 opened the pull request:
When we deploy for RDSS we require two storage service locations to exist: `automated` and `interactive`. Currently, these need to be created manually after each (re)deployment.

This pull request addresses this by building on https://github.com/JiscRDSS/archivematica-storage-service/pull/24 by using the new API added in that PR to automate the creation of these storage locations during the bootstrap stage for the Storage Service container.

This is implemented as a simple Python script that uses the `requests` module (already installed in the Storage Service container) to interact with the SS via its API to check if the required locations exist and if not then create them. The Makefile `bootstrap-storage-service` target invokes this script within the `archivematica-storage-service` container using the `docker-compose exec` command.

Initially I tried to use `curl` and capture the responses in bash, as described by @sevein in https://github.com/JiscRDSS/archivematica-storage-service/pull/24, but this quickly became a soup of different syntaxes as python and curl was wrapped in bash which was wrapped in a docker-compose invocation within a Makefile target. Horrible! Hence why I decided to simplify it by doing everything in Python.

The second commit in this PR is a bit of housekeeping; it makes the way we wait for MySQL to be ready consistent with how it's now done in QA, and also adds a few extra comments, also from the QA config.

I've tested this locally and the storage locations are added as expected, and are visable in the Dashboard Transfer UI. I've also ensured this is idempotent - running the script multiple times won't cause the storage locations to be created multiple times.

lower29 commented:
Good point- I’d forgotten that the API credentials aren’t hard coded for all deployments. I’d also not realised that it isn’t possible to add to the container in ECS. I’ll update the script to take the parameters you suggest, thanks.

sevein commented on compose/qa/etc/rdss-create-storage-locations.py:
```diff
@@ -0,0 +1,136 @@
+                    help='Base URL of Storage Service API to use.')
+parser.add_argument('--api-user', required=True,
+                    help='Username to use when authenticating with the API.')
+parser.add_argument('--api-pass', required=True,
```
I think this should be `--api-key` instead.

lower29 commented:
@sevein I've made some changes based on your feedback:
* Shebang path is now `/usr/bin/env python`. I didn't use `python3` because it's not installed on the `archivematica-storage-service` image, which is the primary target for this script to run.
* I've changed `--api-pass` to `--api-key` and updated the Makefiles that call the script too.
* I've left `requests` in because frankly the `urllib` is horrible by comparison in terms of the syntax required to do the same thing (post JSON with authentication headers), which is why I opted for `requests` in the first place. It's also installed by default in the `archivematica-storage-service` container - again, this is the primary target for this script. I've added a note about it being required to the top pydoc comment, plus added a try-catch to test for the presence of the module prior to attempting to use it, using the `pkg_resources` module to check if the module is available.

Hopefully this is now enough to get the review approved! 🙂

lower29 commented:
Thanks @sevein. We need to wait now until the `v0.4.0` release is done before we can merge.

lower29 commented:
Thanks @sevein. The repo is frozen atm so we need to wait until `v0.4.0` is released before we can merge this.
